client/state/data-layer/wpcom/sites/media/index.js

```diff
@@ -184,9 +188,12 @@ export const deleteMediaSuccess = ( { siteId }, mediaItem ) => ( dispatch ) => {
 	dispatchFluxRemoveMediaItemSuccess( siteId, mediaItem );
 };
 
-export const deleteMediaError = ( { siteId }, error ) => () => {
-	dispatchFluxRemoveMediaItemError( siteId, error );
-};
+export const deleteMediaError = ( { mediaId } ) => [
+	removeNotice( `delete-media-notice-${ mediaId }` ),
+	errorNotice( translate( "We weren't able to delete this media item." ), {
```

Comment:
I have a nit on the wording here: I'd prefer this instead:

```suggestion
	errorNotice( translate( "We were unable to delete this media item." ), {
```

but after all, I'm not a native speaker, so I wouldn't call this a blocking request. 

Maybe @saramarcondes could chime in 😉